examples/moonolith/ex1.cpp

```diff
@@ -0,0 +1,159 @@
+
```

Comment:
Could you add a short explanation what each example does + some sample runs.

Reply:
Done.